### PR TITLE
Added support for dimension based metadata export

### PR DESCRIPTION
**Pull request by j-s-rawat**
### What
US: https://trello.com/c/vSxzIedn/5674-update-dp-cantabular-metadata-exporter-to-support-export-based-on-specific-dimensions

- Update call to new function that supports dimensions based metadata export.
- Update kafak message to support consumption of dimensions 

### How to review

Check that the new message structure matches the one with the CSV exporter
### Who can review

Anyone

### CODE REVIEW

**Comment by PauloPortugal** [comment]
@j-s-rawat we should avoid using the merge strategy on private branches, as it gets messy

File: schema/schema.go [hunk 1]
```diff
@@ -13,7 +13,8 @@ var (
       {"name": "dataset_id",  "type": "string", "default": ""},
       {"name": "edition",     "type": "string", "default": ""},
       {"name": "version",     "type": "string", "default": ""},
-      {"name": "row_count",   "type": "int", "default": 0}
+      {"name": "row_count",   "type": "int", "default": 0},
+      {"name": "dimensions", "type": { "type": "array", "items": "string"}, "default": [] }
```
Comment (PauloPortugal) [inline schema/schema.go:17]: Can you indent this line please?

File: handler/interface.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ type FileManager interface {
 
 type DatasetAPIClient interface {
 	GetVersion(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition, version string) (dataset.Version, error)
+	GetVersionMetadataSelection(ctx context.Context, req dataset.GetVersionMetadataSelectionInput) (*dataset.Metadata, error)
```
Comment (PauloPortugal) [inline handler/interface.go:20]: We don't need to return a pointer... Inefficiency at memory management as this needs to be placed in the heap to be tracked by the GC.
Not your change...

**Review by PauloPortugal — APPROVED** [review]
A small comment about indentation.
it is shame that we are using a function that returns a pointer...

**Review by PauloPortugal — DISMISSED** [review]

**Review by saminahbab — APPROVED** [review]

File: schema/schema.go (hunk 1 above)
Comment (j-s-rawat) [reply schema/schema.go:17]: done

**Review by saminahbab — APPROVED** [review]

**Pull request merged**